icepyx/core/variables.py

```diff
@@ -482,25 +492,59 @@ def append(self, defaults=False, var_list=None, beam_list=None, keyword_list=Non
         To add all variables and paths in ancillary_data
 
         >>> reg_a.order_vars.append(keyword_list=['ancillary_data']) # doctest: +SKIP
+
+        To add variables as a list of explicit paths
+
+        >>> path_list = ['gt1l/heights/delta_time',
+                         'gt1l/heights/dist_ph_across',
+                         'gt1l/heights/dist_ph_along',
+                         'gt1l/heights/h_ph',
+                         'gt1l/heights/lat_ph',
+                         'gt1l/heights/lon_ph']
+        >>> region_a.order_vars.append(path_list=path_list)  # doctest: +SKIP
+
         """
 
-        assert not (
-            defaults == False
-            and var_list == None
-            and beam_list == None
-            and keyword_list == None
-        ), "You must enter parameters to add to a variable subset list. If you do not want to subset by variable, ensure your is2.subsetparams dictionary does not contain the key 'Coverage'."
+        # Check that at least one allowable combination of keywords are set
+        if (
+            not defaults
+            and not (var_list or beam_list or keyword_list)
+            and not path_list
+        ):
+            raise ValueError(
+                "Either default or path_list, or at least one of var_list, "
+                "beam_list or keyword_list must be set\n"
+                "If you do not want to subset by variable, "
+                "ensure your is2.subsetparams dictionary does "
+                "not contain the key 'Coverage'."
```

Comment:
Thanks for this update to a ValueError!

Given the Variables module is used by Query, Read, and set up to work independently, I'm wondering if it might be confusing to include a "query-specific" component to the warning. Thoughts?

Reply:
Good point!  Should I delete these lines (517 to 519)?

Does the `Variables.append` method get called "under-the-hood" or would it always be called through `order_vars` or a `Variables` object.  That said a warning `ValueError("Either default or path_list, or at least one of var_list, beam_list or keyword_list must be set")` seems explicit and clear enough.

Reply:
If memory serves, `.append` is called under the hood during query or read to ensure the user has the "required" [metadata] variables (I don't recall the exact list, but for subsetting it makes sure things like delta_time are there, and for read it adds things like spot/gt that icepyx needs to parse data into xarray). But even that is done on an instance of `Variables` via `query.order_vars` or `read.vars`.

>a warning ValueError("Either default or path_list, or at least one of var_list, beam_list or keyword_list must be set") seems explicit and clear enough.

Agreed.
> Should I delete these lines (517 to 519)?

Sounds good!

Reply:
I removed the lines indicated above and pushed.